pulser-core/pulser/sequence/_schedule.py

```diff
@@ -177,6 +219,34 @@ def find_slm_mask_times(self) -> list[int]:
                 break
         return mask_time
 
+    def enable_eom(
+        self,
+        channel_id: str,
+        amp_on: float,
+        detuning_on: float,
+        detuning_off: float,
+    ) -> None:
+        channel_obj = self[channel_id].channel_obj
+        if any(isinstance(op.type, Pulse) for op in self[channel_id]):
+            # Wait for the last pulse to ramp down (if needed)
+            self.wait_for_fall(channel_id)
+            # Account for time needed to ramp to desired amplitude
+            self.add_delay(2 * channel_obj.rise_time, channel_id)
```

Comment:
Where does the factor 2 come from here?

Reply:
In fact, by definition the `rise_time` is the time it takes to ramp from 10% to 90% of the maximum value. Visually, I find that roughly `2 * rise_time` is what it takes to go from 0% to 100%.

Reply:
I'll add a comment here too 😅